File: Lab1/Cuadratica/src/prac5/Cuadratica.java
package prac5;

import java.util.Scanner;

public class Cuadratica {

	public static double getDiscriminante(double a, double b, double c) {
		return b * b - 4 * a * c;
	}

	public static Double getRaiz1(double a, double b, double discriminante) {
		if (discriminante < 0)
			return null;
		return (-b + Math.sqrt(discriminante)) / (2 * a);
	}

	public static Double getRaiz2(double a, double b, double discriminante) {
		if (discriminante < 0)
			return null;
		return (-b - Math.sqrt(discriminante)) / (2 * a);
	}

	public static void resolverEcuacion(double a, double b, double c) {
		double discriminante = getDiscriminante(a, b, c);

		if (discriminante > 0) {
			System.out.printf("La ecuación tiene dos raíces %.6f y %.6f%n", getRaiz1(a, b, discriminante),
					getRaiz2(a, b, discriminante));
		} else if (discriminante == 0) {
			System.out.printf("La ecuación tiene una raíz %.0f%n", getRaiz1(a, b, discriminante));
		} else {
			System.out.println("La ecuación no tiene raíces reales");
		}
	}

	public static void main(String[] args) {
		Scanner scanner = new Scanner(System.in);

		System.out.print("Ingrese a, b, c: ");
		double a = scanner.nextDouble();
		double b = scanner.nextDouble();
		double c = scanner.nextDouble();

		resolverEcuacion(a, b, c);
		scanner.close();
	}
}

File: Lab1/cuadratica.py
import math

class Cuadratica:
    def __init__(self, a, b, c):
        self.__a = a
        self.__b = b
        self.__c = c

    def getDiscriminante(self):
        return self.__b ** 2 - 4 * self.__a * self.__c

    def getRaiz1(self):
        d = self.getDiscriminante()
        if d < 0:
            return None
        return (-self.__b + math.sqrt(d)) / (2 * self.__a)

    def getRaiz2(self):
        d = self.getDiscriminante()
        if d < 0:
            return None
        return (-self.__b - math.sqrt(d)) / (2 * self.__a)

    def resolver(self):
        d = self.getDiscriminante()
        if d > 0:
            print(f"La ecuación tiene dos raíces {self.getRaiz1():.6f} y {self.getRaiz2():.6f}")
        elif d == 0:
            print(f"La ecuación tiene una raíz {self.getRaiz1():.0f}")
        else:
            print("La ecuación no tiene raíces reales")

if __name__ == "__main__":  
    a, b, c = map(float, input("Ingrese a, b, c: ").split())
    ecuacion = Cuadratica(a, b, c)
    ecuacion.resolver()
